restli-common/src/main/java/com/linkedin/restli/common/ComplexResourceKey.java

```diff
@@ -227,11 +242,13 @@ public void makeReadOnly()
    *
    * @param keyDataMap untyped DataMap - all primitive values are represented as strings.
    * @param complexKeyType type of {@link ComplexResourceKey}
+   * @param throwErrorOnValidationFailure if set true throws IllegalArgumentException on validation failure
    * @return {@link ComplexResourceKey} initialized with id and param values specified in
    *         the input DataMap
    */
   public static ComplexResourceKey<RecordTemplate, RecordTemplate> buildFromDataMap(DataMap keyDataMap,
-                                                                                    ComplexKeySpec<?, ?> complexKeyType)
+                                                                                    ComplexKeySpec<?, ?> complexKeyType,
+                                                                                    boolean throwErrorOnValidationFailure)
```

Comment:
`throwErrorOnValidationFailure` is kind of a mouthful... I'd suggest renaming to something like `enforceValidation`

Reply:
done